internal/trace/transport.go

```diff
@@ -26,31 +28,45 @@ import (
 // request and add hooks to report HTTP tracing events.
 type Transport struct {
 	http.RoundTripper
+	mu    sync.Mutex
+	count uint64
 }
 
 func NewTransport(base http.RoundTripper) *Transport {
-	return &Transport{base}
+	return &Transport{base, sync.Mutex{}, 0}
 }
 
 // RoundTrip calls base roundtrip while keeping track of the current request.
 func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	atomic.AddUint64(&t.count, 1)
```

Comment:
```suggestion
	count := atomic.AddUint64(&t.count, 1)
```
Request or response logging in this roundtrip call should be based on the local variable `count`

Reply:
changed accordingly